tss/rsa/rsa_threshold.go

```diff
@@ -143,18 +150,32 @@ func Deal(randSource io.Reader, players, threshold uint, key *rsa.PrivateKey, ca
 		}
 	}
 
+	var groupKey *big.Int
+	if hasSafePrimes {
+		groupKey, err = qndleq.SampleQn(randSource, key.N)
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	shares := make([]KeyShare, players)
 
 	// 1 <= i <= l
 	for i := uint(1); i <= players; i++ {
 		shares[i-1].Players = players
 		shares[i-1].Threshold = threshold
 		// Σ^{k-1}_{i=0} | a_i * X^i (mod m)
-		poly := computePolynomial(threshold, a, i, &m)
-		shares[i-1].si = poly
+		si := computePolynomial(threshold, a, i, &m)
+		shares[i-1].si = si
 		shares[i-1].Index = i
-		if cache {
```

Comment:
nit: If caching is no longer optional, documentation should be update here https://github.com/cloudflare/circl/blob/tssRSAverif/tss/rsa/keyshare.go#L30

Reply:
also the if check should be removed here: https://github.com/cloudflare/circl/blob/5dcac578945a71dc912e27babc46148672c9e61c/tss/rsa/keyshare.go#L73C2-L73C2

Reply:
done